src/dialects/oracle/transaction.js

```diff
@@ -40,19 +40,24 @@ export default class Oracle_Transaction extends Transaction {
     const t = this
     return Promise.try(() =>
       config.connection || t.client.acquireConnection()
-    ).tap(connection => {
-      if (!t.outerTx) {
-        connection.setAutoCommit(false)
-      }
-    }).disposer(connection => {
-      debugTx('%s: releasing connection', t.txid)
-      connection.setAutoCommit(true)
-      if (!config.connection) {
-        t.client.releaseConnection(connection)
-      } else {
-        debugTx('%s: not releasing external connection', t.txid)
-      }
-    })
+    )
+      .then(connection => {
+        connection.__knexTxId = this.txid;
```

Comment:
Not sure why this is not working, unable to test locally..

Reply:
I will setup docker, which runs oracle tests locally in container. node-oracledb driver is pretty painful to install on every platform 😬

Reply:
Yeah I tried installning on Windows, and I can vouch for the difficulty... 😥

Reply:
I don't see anything wrong either. I suppose this.txid is not set at this point. looks like in other places they are referring it with t.txid, but it shouldn't matter. Will be easier to figure out when tests can be ran locally.

Reply:
https://github.com/tgriesser/knex/pull/2491 there is PR for adding npm script to run oracledb tests locally in docker container

Reply:
Took more time than I care to admit to find the issue. Turns out I only changed in `dialects/oracle/transaction.js` assuming that was what test was running, but it was actually running `dialects/oracledb/transaction.js` which I had forgotten to change